### ISSUE TITLE
[Good first issue] [API CONFORMANCE] [CPU PLUGIN] Support unsupported property OPTIMAL_BATCH_SIZE in CPU plugin

### ISSUE THREAD

**Issue by tiger100256-hu** [openvinotoolkit/openvino#22118]
## Description:
API conformance suite is OV validation tool checks a plugin conditions from API implementation perspective.
`ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps.ChangeCorrectDeviceProperties/target_device=CPU_properties=*`  fails on `CPU` Plugin

### RO (ReadOnly):
* OPTIMAL_BATCH_SIZE

## How to reproduce:
1. Build the OV using `-DENABLE_TESTS=ON -DENABLE_FUNACTIONAL_TESTS=ON -DENABLE_INTEL_CPU=ON`
2. Build `ov_api_conformance_tests` target
```
 ./ov_api_conformance_tests --gtest_filter="ov_plugin/OVCheckMetricsPropsTests_ModelD
ependceProps.ChangeCorrectDeviceProperties/target_device=CPU_properties={OPTIMAL_BATCH_SIZE*"
Note: Google Test filter = ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps.ChangeCorrectDeviceProperties/target_device=CPU_properties={OPTIM
AL_BATCH_SIZE*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps
[ RUN      ] ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps.ChangeCorrectDeviceProperties/target_device=CPU_properties={OPTIMAL_BATCH_SIZE:
}

MEM_USAGE=47184KB
[ CONFORMANCE ] Influence coefficient: 0
src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:456: Failure
Value of: supported
  Actual: false
Expected: true
property is not supported: MODEL_PTR
[  FAILED  ] ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps.ChangeCorrectDeviceProperties/target_device=CPU_properties={OPTIMAL_BATCH_SIZE:
}, where GetParam() = ("CPU", { ("OPTIMAL_BATCH_SIZE", ) }) (5 ms)
[----------] 1 test from ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps (5 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (5 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ov_plugin/OVCheckMetricsPropsTests_ModelDependceProps.ChangeCorrectDeviceProperties/target_device=CPU_properties={OPTIMAL_BATCH_SIZE:
}, where GetParam() = ("CPU", { ("OPTIMAL_BATCH_SIZE", ) })

 1 FAILED TEST
```

## How to fix
support OPTIMAL_BATCH_SIZE in CPU plugin, set 1 as the default value, return the default value. Example PR https://github.com/openvinotoolkit/openvino/pull/22007

## Useful links:
* [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
* [Conformance readme](https://github.com/openvinotoolkit/openvino/tree/master/src/tests/functional/plugin/conformance/test_runner)
* [CPU plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/README.md)
* [OV CONTRIBUTION GUIDE](https://github.com/openvinotoolkit/openvino/tree/master/?tab=readme-ov-file#how-to-contribute)
* [PROPERIES Description](https://docs.openvino.ai/2023.2/groupov_runtime_cpp_prop_api.html)
* [OPTIMAL_BATCH_SIZE](https://github.com/openvinotoolkit/openvino/blob/7c37e0916c6cc2c4df14877c10346a9cbbf844bd/src/inference/include/openvino/runtime/properties.hpp#L687)

## Contacts:
* @tiger100256-hu

**Comment by karanjakhar** [comment]
.take

**Comment by yuxu42** [comment]
Hi @karanjakhar, are you still working on it? And let us know if you need any help.

**Comment by karanjakhar** [comment]
Hi @yuxu42 yes I am working on it.

**Comment by p-wysocki** [comment]
Hi @karanjakhar, are there any updates on this? Do you have any questions, anything we can help you with?

**Comment by karanjakhar** [comment]
Hi @p-wysocki 
Sorry for the delay. I have started working on it. This weekend I will get back to it.

**Comment by Hunter-bitbit** [comment]
Hello! May I also work on this problem? This is my first problem and I want to try contributing! Can you help me with reproducing the error?

**Comment by mlukasze** [comment]
hey @Hunter-bitbit
this one is taken, but maybe you would be interested in different one we have in a backlog?
https://github.com/orgs/openvinotoolkit/projects/3
Starting next week we will have more to offer, again :)

**Comment by Hunter-bitbit** [comment]
Will do @mlukasze!

**Comment by iefode** [comment]
@Hunter-bitbit I will create tickets for Operation conformance soon. You can take them :)
Thanks!

**Comment by karanjakhar** [comment]
Hi @tiger100256-hu , @p-wysocki 
Please help me: https://github.com/openvinotoolkit/openvino/compare/master...karanjakhar:openvino:issue_22118 
I have these changes so far. It's not working. I followed the example PR.

**Comment by iefode** [comment]
@karanjakhar 
Could you please update your branch to be updated with the latest master branch? You have merge conflicts... Please, create a P, we will review the changes
Also, please remove the fixed test from [`API_expected_failures`](https://github.com/openvinotoolkit/openvino/blob/master/src/tests/test_utils/functional_test_utils/layer_tests_summary/github/skip_configs/CPU/expected_failures_API.csv) as in https://github.com/openvinotoolkit/openvino/pull/22227/files

Changes look correct.

**Comment by tiger100256-hu** [comment]
@karanjakhar  sorry for late reply, I think the **ov::optimal_batch_size** depend **on ov::hint:model**  https://github.com/openvinotoolkit/openvino/blob/3986f55da28ee343780b623621d6b531a38af928/src/inference/include/openvino/runtime/properties.hpp#L455, so also need to support ov::hint::model in CPU plugin.  could you also help to add it ? thanks
https://github.com/openvinotoolkit/openvino/blob/3986f55da28ee343780b623621d6b531a38af928/src/plugins/auto/src/auto_compiled_model.cpp#L157

**Comment by karanjakhar** [comment]
Hi @tiger100256-hu  
Yes, I would like to work on that. I will work on it this weekend.

**Comment by tiger100256-hu** [comment]
@karanjakhar  I see you created PR 23199 and closed it last week, If you need any help, feel free to let us know

**Comment by karanjakhar** [comment]
Hi @tiger100256-hu 
Today I will update it. Apologies for the delay.

**Comment by p-wysocki** [comment]
Hi @karanjakhar, please link the PR in the Development section once you open it: 
![image](https://github.com/openvinotoolkit/openvino/assets/55858107/a619b36b-998b-496f-aa55-100a38812d9c)

**Comment by p-wysocki** [comment]
I'm reopening the issue due to current assignee's inactivity. @karanjakhar please let us know if you're still working on this.

**Comment by karanjakhar** [comment]
@p-wysocki Yes I have code ready. Sorry due to some other work couldn't complete it. This weekend I will push it.

**Comment by kevinzhangc** [comment]
#WLB#+ .take

**Comment by iefode** [comment]
Hello @kevinzhangc!

What is the status of the issue? Please, let us know in case of any issues

**Comment by tiger100256-hu** [comment]
no activity on this issue, so create a PR https://github.com/openvinotoolkit/openvino/pull/28948 to fix the issue, close it
